src/main/java/ui/TestController.java

```diff
@@ -79,4 +85,46 @@ public static boolean isCloseOnQuit() {
         return hasUI() && commandLineArgs.getOrDefault("closeonquit", "false").equalsIgnoreCase("true");
     }
 
+    /**
+     * Creates a Preferences that store data in test config file if run with test mode
+     * or in a default config file specified in the Preferences class
+     * @return
+     */
+    public static Preferences createApplicationPreference() {
+        if (isTestMode()) {
+            return createTestPreferences();
+        }
+
+        return new Preferences(Preferences.GLOBAL_CONFIG_FILE);
+    }
+
+    /**
+     * Creates a Preferences that stores data in the test directory
+     * @return
+     */
+    public static Preferences createTestPreferences() {
+        return new Preferences(Preferences.TEST_CONFIG_FILE);
+    }
+
+    /**
+     * Creates a RepoIO for the application that uses different components
+     * depending on various test options: --test, --testjson etc.
+     * @return
+     */
+    public static RepoIO createApplicationRepoIO() {
+        return new RepoIO(isTestMode() ? Optional.of(new DummySource()) : Optional.empty(),
+                          isTestJSONEnabled() ? Optional.of(new JSONStoreStub()) : Optional.empty(),
```

Comment:
I think from the original `RepoIO`, when `isTestJSONEnabled` is true _and_ is in test mode, it will use true JSON. If `isTestJSONEnabled` is false _and_ is in test mode, then it will use the stub (`isTestJSONEnabled` signifies that we will use some real JSON operation in test folder).